### PR TITLE
Bottom resizable pane

### PR DESCRIPTION
**Pull request by stMerlHin**
I built a BottomResizable  pane widget , an implementation of [apple HSplitView](https://developer.apple.com/design/human-interface-guidelines/components/layout-and-organization/split-views) base on the existing ResizablePane, addressed [#344](https://github.com/GroovinChip/macos_ui/issues/344).
A little video which shows the widget (consider using firefox or chorme if the video is not palying).

[Capture vidéo du 2023-02-08 02-21-41.webm](https://user-images.githubusercontent.com/77164238/217412466-8c45656e-969c-4a84-ba6d-6b9b07764881.webm)

## Pre-launch Checklist

- [x] I have run `dartfmt` on all changed files 
- [x] I have incremented the package version as appropriate and updated `CHANGELOG.md` with my changes 
- [x] I have added/updated relevant documentation 
- [x] I have run "optimize/organize imports" on all changed files
- [x] I have addressed all analyzer warnings as best I could

### CODE REVIEW

**Comment by GroovinChip** [comment]
@stMerlHin Before I review, it occurs to me to wonder if this could be baked into `ResizablePane` and adding a `ResizableSide.bottom` to the `ResizableSide` enum. What do you think?

**Comment by stMerlHin** [comment]
I was thinking so. The bottom resizable pane use almost the same base code, I was tempted to inherit it from resizable pane and then do some overrides because I was afraid to touch an existing and functional widget. It will  pretty good to have one resizable widget with capability to make it horizontal or vertical.

Also the `ResizableSide` should be `ResizableSide.top` instead of `ResizableSide.bottom`

**Comment by GroovinChip** [comment]
Great! Do you mind trying to go that route, then? I really appreciate your work on this 🙏 

And you're correct, it should be `ResizableSide.top`, not bottom.

**Comment by stMerlHin** [comment]
I'm very pleased to contribute to my favorite package. will do so and push the new version very soon.

Potential issue, the `MacosScaffold` takes as children `ContentArea` and `ResizablePane` so I'm asking myself if adding capability to set the `ResizableSide` to `top` will not end up with weird behavior since the `MacosScaffold` arranged his children as row.

**Comment by GroovinChip** [comment]
> I'm very pleased to contribute to my favorite package. will do so and push the new version very soon.
> 
> 
> 
> Potential issue, the `MacosScaffold` takes as children `ContentArea` and `ResizablePane` so I'm asking myself if adding capability to set the `ResizableSide` to `top` will not end up with weird behavior since the `MacosScaffold` arranged his children as row.

It's certainly a good question. If it doesn't work out, at least we've got the work you've already done. I appreciate you trying!

**Comment by stMerlHin** [comment]
I've finished adding capablity the `ResizablePane` 

> @stMerlHin Before I review, it occurs to me to wonder if this could be baked into `ResizablePane` and adding a `ResizableSide.bottom` to the `ResizableSide` enum. What do you think?

Hey @GroovinChip  I've done adding the feature. Thanks.

**Comment by GroovinChip** [comment]
> I've finished adding capablity the `ResizablePane`
> 
> > @stMerlHin Before I review, it occurs to me to wonder if this could be baked into `ResizablePane` and adding a `ResizableSide.bottom` to the `ResizableSide` enum. What do you think?
> 
> Hey @GroovinChip I've done adding the feature. Thanks.

Amazing!!! Thanks so much! I'll do my best to review today.

**Comment by stMerlHin** [comment]
Nice, I'll do every modification if needed. Thanks.

**Comment by stMerlHin** [comment]
Nice, I'll do any modification if needed. Thanks.

File: lib/src/layout/resizable_pane.dart [hunk 1]
```diff
@@ -61,19 +64,28 @@ class ResizablePane extends StatefulWidget {
   /// resizable side of this widget.
   final bool isResizable;
 
-  /// Specifies the maximum width that this [ResizablePane] can have.
+  /// Specifies the maximum width or height that this [ResizablePane] can have
+  /// according to his orientation (the orientation is horizontal if the
+  /// [resizableSide] is [ResizableSide.left] or [ResizableSide.right] and
+  /// vertical if the [resizableSide] is [ResizableSide.top]).
```
Comment (GroovinChip) [inline lib/src/layout/resizable_pane.dart]:
```suggestion
  /// Specifies the maximum width or height that this [ResizablePane] can have
  /// according to its orientation. 
  /// 
  /// The orientation is horizontal if the [resizableSide] is 
  /// [ResizableSide.left] or [ResizableSide.right] and vertical if the 
  /// [resizableSide] is [ResizableSide.top]).
```

File: lib/src/layout/resizable_pane.dart [hunk 2]
```diff
@@ -61,19 +64,28 @@ class ResizablePane extends StatefulWidget {
...
   ///
   /// The value can be null and defaults to `500.0`.
```
Comment (GroovinChip) [inline lib/src/layout/resizable_pane.dart]:
```suggestion
  /// If this value is null, it defaults to `500.0`.
```

File: lib/src/layout/resizable_pane.dart [hunk 3]
```diff
@@ -61,19 +64,28 @@ class ResizablePane extends StatefulWidget {
...
-  final double maxWidth;
+  final double maxSize;
 
-  /// Specifies the minimum width that this [ResizablePane] can have.
-  final double minWidth;
+  /// Specifies the minimum width of height that this [ResizablePane] can have
+  /// according to his orientation (the orientation is horizontal if the
+  /// [resizableSide] is [ResizableSide.left] or [ResizableSide.right] and
+  /// vertical if the [resizableSide] is [ResizableSide.top]).
```
Comment (GroovinChip) [inline lib/src/layout/resizable_pane.dart]:
```suggestion
  /// Specifies the minimum width of height that this [ResizablePane] can have
  /// according to its orientation. 
  /// 
  /// The orientation is horizontal if the [resizableSide] is 
  /// [ResizableSide.left] or [ResizableSide.right] and vertical if the 
  /// [resizableSide] is [ResizableSide.top].
```

File: lib/src/layout/resizable_pane.dart [hunk 4]
```diff
@@ -61,19 +64,28 @@ class ResizablePane extends StatefulWidget {
...
+  final double minSize;
 
-  /// Specifies the width that this [ResizablePane] first starts width.
+  /// Specifies the width or height that this [ResizablePane] first starts with
+  /// according to his orientation (the orientation is horizontal if the
+  /// [resizableSide] is [ResizableSide.left] or [ResizableSide.right] and
+  /// vertical if the [resizableSide] is [ResizableSide.top]).
```
Comment (GroovinChip) [inline lib/src/layout/resizable_pane.dart]:
```suggestion
  /// Specifies the width or height that this [ResizablePane] first starts with
  /// according to its orientation. 
  /// 
  /// The orientation is horizontal if the [resizableSide] is 
  /// [ResizableSide.left] or [ResizableSide.right] and vertical if the 
  /// [resizableSide] is [ResizableSide.top]).
```

File: test/layout/resizeable_pane_test.dart [hunk 5]
```diff
@@ -7,150 +7,226 @@ void main() {
...
-
-          await tester.drag(
-            dragFinder,
-            Offset(safeDelta, 0),
+      bool verticallyResizable = side == ResizableSide.top;
+
+      group(
+        side == ResizableSide.top
+            ? "top"
+            : (side == ResizableSide.left ? "left" : "right"),
```
Comment (GroovinChip) [inline test/layout/resizeable_pane_test.dart]:
```suggestion
            ? 'top'
            : (side == ResizableSide.left ? 'left' : 'right'),
```

File: test/layout/resizeable_pane_test.dart [hunk 6]
```diff
@@ -7,150 +7,226 @@ void main() {
...
+          final resizablePaneFinder = find.byWidget(resizablePane);
+          final dragFinder = find.descendant(
+            of: resizablePaneFinder,
+            matching: find.byType(GestureDetector),
           );
-        });
 
-        testWidgets('dragging wider respects maxWidth', (tester) async {
-          await tester.pumpWidget(view);
+          //No need to check if the resizable side is top because directionModifier
```
Comment (GroovinChip) [inline test/layout/resizeable_pane_test.dart]:
```suggestion
          // No need to check if the resizable side is top because directionModifier
```

**Review by GroovinChip — DISMISSED** [review]

**Review by GroovinChip — APPROVED** [review]
LGTM pending all checks pass 🎉

**Pull request merged**